open3d_slam_ros/include/open3d_slam_ros/SlamMapInitializer.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * SlamMapInitializer.hpp
+ *
+ *  Created on: Jun 16, 2022
+ *      Author: lukaszpi
+ */
+
+#pragma once
+
+#include <memory>
+#include <atomic>
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <nav_msgs/Odometry.h>
+#include <geometry_msgs/PoseWithCovarianceStamped.h>
+#include "open3d_slam_ros/Parameters.hpp"
+#include "open3d_slam/SlamWrapper.hpp"
+#include <visualization_msgs/InteractiveMarker.h>
+#include <visualization_msgs/InteractiveMarkerFeedback.h>
+#include <interactive_markers/interactive_marker_server.h>
+#include <interactive_markers/menu_handler.h>
+#include <std_srvs/Trigger.h>
+
+namespace o3d_slam {
+
+class SlamMapInitializer{
```

Comment:
This class I would also split into ROS and non ros. For non-ros initialization, you can always just provide the initial pose in the config file. Then if you wanna do it at runtime the way we are doing it, then you use ROS. It should be the next PR, just to limit the complexity.